src/olympia/versions/models.py

```diff
@@ -313,8 +313,11 @@ def delete(self, hard=False):
             self.deleted = True
             self.save()
 
-        for preview in previews:
-            delete_preview_files.delay(preview)
+        for preview_pk in previews_pks:
+            # FIXME: oops that's a problem, the hard delete will have deleted
+            # the preview, so we can't get it from the db to delete the files.
+            # Need to be smarter.
+            delete_preview_files.delay(preview_pk)
```

Comment:
True, though do we actually hard delete versions anywhere outside of tests?

Reply:
Good point. I guess it's not that bad... still need to handle that somehow, it's breaking tests.

Reply:
I should be able to work around this issue by just not calling the task in hard deletion, letting the `post_delete` signal be triggered and handle that for us.